test/suite/test_chunkcache04.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, sys
+import wiredtiger, wttest
+
+from wtdataset import SimpleDataSet
+from wtscenario import make_scenarios
+from time import sleep
+
+
+'''
+- Functional testing for the ingesting. Verify ingests are taking place with both pinned and unpinned chunks.
+- Verify that when ingesting new chunks with old pinned objects, we are releasing the pin on the old objects. 
+'''
+class test_chunkcache4(wttest.WiredTigerTestCase):
+    rows = 10000
+
+    format_values = [
+        ('column', dict(key_format='r', value_format='S')),
+        ('row_string', dict(key_format='S', value_format='S')),
+    ]
+
+    cache_types = [('in-memory', dict(chunk_cache_type='DRAM'))]
+    if sys.byteorder == 'little':
+        # WT's filesystem layer doesn't support mmap on big-endian platforms.
+        cache_types.append(('on-disk', dict(chunk_cache_type='FILE')))
+
+    scenarios = make_scenarios(format_values, cache_types)
+
+    def conn_config(self):
+        if not os.path.exists('bucket2'):
+            os.mkdir('bucket2')
+
+        return 'tiered_storage=(auth_token=Secret,bucket=bucket2,bucket_prefix=pfx_,name=dir_store),' \
+            'chunk_cache=[enabled=true,chunk_size=512KB,capacity=20MB,pinned=("table:chunkcache01", "table:chunkcache02"),type=DRAM,storage_path=WiredTigerChunkCache]'
```

Comment:
Shouldn't this use the cache type from the scenario?

Reply:
Yup this is a good spot I took it out for testing.

Reply:
[d0f1554](https://github.com/wiredtiger/wiredtiger/pull/9752/commits/d0f1554a1a4b797a5461c495bbecad346372b776)